feat: APP-366 account orders

blushi opened the pull request:
## Description

https://regennetwork.atlassian.net/browse/APP-366


<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

<!-- This should include steps on how to test the fixes or features within the marketplace app and/or the website.
If applicable, this should also include links to the created/updated components on storybook. -->

From https://deploy-preview-2566--regen-marketplace.netlify.app/
1. First, login and buy some credits with fiat and with crypto (older crypto orders below a certain block height are not available on staging because of state pruning on our redwood node, this won't be the case on mainnet since we use the archive node where we should be able to see old crypto orders)
2. Check out "my orders" and verify orders info there

~~**NB: credit card info is missing for now, I've realized that we need some server update for that, which I'm working on, but the rest can still be tested in the mean time.**~~ DONE

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

blushi commented:
@erikalogie @S4mmyb see testing instructions

erikalogie commented:
Some styling and other comments:
<img width="1010" alt="Monosnap Regen Marketplace 2024-12-05 10-26-38" src="https://github.com/user-attachments/assets/d3f5840b-261d-4a1e-9e12-ef021014c4f6">

<img width="1290" alt="Monosnap Regen Marketplace 2024-12-05 10-54-06" src="https://github.com/user-attachments/assets/19915eca-e4cf-4152-8e14-c7b733febbf1">

<img width="423" alt="Monosnap Regen Marketplace 2024-12-05 10-28-52" src="https://github.com/user-attachments/assets/24e62013-7e2e-4f58-b017-149895b4513d">

<img width="852" alt="Monosnap Regen Marketplace 2024-12-05 10-58-46" src="https://github.com/user-attachments/assets/44031fff-0b74-4b19-a961-702c503b51da">

On another notes, do we need a task to have the name of the user account change to the name that I put in the 2nd step of the buy flow?

blushi commented:
> On another notes, do we need a task to have the name of the user account change to the name that I put in the 2nd step of the buy flow?

We have one: https://regennetwork.atlassian.net/browse/APP-441

blushi commented:
@erikalogie you can have another look, I've also added support for displayed credit card info.

erikalogie commented:
A few more small alignment things:

<img width="446" alt="Monosnap Regen Marketplace 2024-12-09 10-22-28" src="https://github.com/user-attachments/assets/4751ed7f-15a9-4df1-97ec-9f7bc1ddc7e2">
<img width="494" alt="Monosnap Screenshot 2024-12-09 at 10 19 05 AM 2024-12-09 10-19-37" src="https://github.com/user-attachments/assets/e974b8e3-2bdb-4650-aa32-f90ae7972a83">

blushi commented:
> 
> <img alt="Monosnap Regen Marketplace 2024-12-09 10-22-28" width="446" src="https://private-user-images.githubusercontent.com/8460237/393870124-4751ed7f-15a9-4df1-97ec-9f7bc1ddc7e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NTQ2NDcsIm5iZiI6MTczMzc1NDM0NywicGF0aCI6Ii84NDYwMjM3LzM5Mzg3MDEyNC00NzUxZWQ3Zi0xNWE5LTRkZjEtOTdlYy05ZjdiYzFkZGM3ZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMTQyNTQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWFlODE4ODJmZGEzNjM4ZWIxOWEwNTE0YmJiOGY5NGUyZTBlMmY5ZTA0ODlhNTI3Zjc2ODhhZjJiNGQ2YjFiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.76vodgufMxSjx1I3iu9tbBN-AAhjajvsVhE0Fc6CtfU"> <img alt="Monosnap Screenshot 2024-12-09 at 10 19 05 AM 2024-12-09 10-19-37" width="494" src="https://private-user-images.githubusercontent.com/8460237/393870139-e974b8e3-2bdb-4650-aa32-f90ae7972a83.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NTQ2NDcsIm5iZiI6MTczMzc1NDM0NywicGF0aCI6Ii84NDYwMjM3LzM5Mzg3MDEzOS1lOTc0YjhlMy0yYmRiLTQ2NTAtYWEzMi1mOTBhZTc5NzJhODMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMTQyNTQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdhYjBiOTI3ZjY3NTUyZmQ5ZGQyZDA2MmVhYzNmN2I3MGI5MDYzODhiMjhhZDg3MzZmOTJhNDAwN2U5YzFhMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.lhWF2b-KUrSH4Uy6KfQsRnmb0sSSdYcXvsJ7EP-cPTc">

@erikalogie how much should I increase the space between image and title on mobile? there's a space of 10px as on figma

blushi commented:
@erikalogie I'm asking about this:
![image](https://github.com/user-attachments/assets/d46fecc5-65f1-4c3f-abed-0b15e6972ebd)

erikalogie commented:
> @erikalogie I'm asking about this: ![image](https://private-user-images.githubusercontent.com/3149358/393874109-d46fecc5-65f1-4c3f-abed-0b15e6972ebd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NTUxNTEsIm5iZiI6MTczMzc1NDg1MSwicGF0aCI6Ii8zMTQ5MzU4LzM5Mzg3NDEwOS1kNDZmZWNjNS02NWYxLTRjM2YtYWJlZC0wYjE1ZTY5NzJlYmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMTQzNDExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjUwZWM3MzZhYmY5ODRmMjcxYmFjNTUzMDdjMDdkMGEwY2U5ODNmYjBmODM4NzcyNGM2Zjc4MmU1NDljYzY0ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.UYO46OlLMVWeh-OV4EdhFV3fENpLKfrWyqubFS0ySKM)

Oh maybe that part will look fine if the space below is changed. So leave it for now.

blushi commented:
@erikalogie you can have another look
Although we need to wait on https://github.com/regen-network/regen-server/pull/517 to be merged before testing the stripe receipt, that I've implemented (not yet pushed)
I'll create a follow-up task to implement the custom receipt page for crypto purchases

erikalogie commented:
Everything looks good, just let me know when the server PR is merged and I can test the stripe receipt button

erikalogie commented:
LGTM! Nice work here

blushi commented:
@r41ph I've added some new stuff since your initial approval, so let me know if you wanna have another look before I merge this, thanks!

r41ph commented:
@blushi  the code looks good to me. A couple of things:

- There are tests failing in`Order.test.tsx` and `CreditsAmount.test.tsx` (I've fixed some test in CreditsAmount here: https://github.com/regen-network/regen-web/pull/2572 after the latest changes and think you can probably ignore those for now)
- The buttons on mobile should go in the same line and there is some spacing differences between the order details as well as per figma designs (https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=1229-75403&node-type=frame&m=dev):

Storybook
![image](https://github.com/user-attachments/assets/72d95e1a-461f-424c-a5cb-13b7683a096c)


Figma
![image](https://github.com/user-attachments/assets/d92bd906-f60c-4041-a8ac-17db3f40ebf3)

blushi commented:
> @blushi the code looks good to me. A couple of things:
> 
> * There are tests failing in`Order.test.tsx` and `CreditsAmount.test.tsx` (I've fixed some test in CreditsAmount here: [fix APP 407 inputs logic #2572](https://github.com/regen-network/regen-web/pull/2572) after the latest changes and think we can probably ignore those for now)
> * The buttons on mobile should go in the same line and there is some spacing differences in the order details as well as per figma designs (https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=1229-75403&node-type=frame&m=dev):
> 
> Storybook ![image](https://private-user-images.githubusercontent.com/20154088/395157689-72d95e1a-461f-424c-a5cb-13b7683a096c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxNjYzMDAsIm5iZiI6MTczNjE2NjAwMCwicGF0aCI6Ii8yMDE1NDA4OC8zOTUxNTc2ODktNzJkOTVlMWEtNDYxZi00MjRjLWE1Y2ItMTNiNzY4M2EwOTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA2VDEyMjAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MmFhNjE1NjlhNWRiZGU2NmZlYWFhMTc2YWQ0NGY0YTUyNzhlMmUxYmVhMTlmZjU4YzcwMDZkOGQ3ZTRkN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oMtM3gIykNNtZha_EUFrR1rEbD8dc6cgp660P6BKR70)
> 
> Figma ![image](https://private-user-images.githubusercontent.com/20154088/395157810-d92bd906-f60c-4041-a8ac-17db3f40ebf3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYxNjYzMDAsIm5iZiI6MTczNjE2NjAwMCwicGF0aCI6Ii8yMDE1NDA4OC8zOTUxNTc4MTAtZDkyYmQ5MDYtZjYwYy00MDQxLWE4YWMtMTdkYjNmNDBlYmYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA2VDEyMjAwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkYmIzMjg2OWUyNTIyNWE1Zjc0MDRiM2E0OTA3OWRmOWEzZTMyYjY0ZTBjYmM1YmU4MzdmZTg1YWE1NzcxOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.o0zqG_UK_XHjwZd5KDXl8ufTx8VUC2Fp7qy6xnS4BI4)

I've fixed the Order tests and spacing issues. Could you have another look? @r41ph 
For mobile devices with width > 380px, the buttons get displayed on the same line as in the designs. 

![image](https://github.com/user-attachments/assets/f264b220-2617-4f0d-97e1-b22ebd580bb7)

Else (like it's the case on your screen shot), they wrap and this prevents overflowing on even smaller screens, which would look bad:

![image](https://github.com/user-attachments/assets/105414f5-9d5d-4ca9-9dba-907fc25f3b64)


We would need to set a lower horizontal padding for the buttons which would look quite bad as well IMO:
![image](https://github.com/user-attachments/assets/60f1a3de-fdf2-40f9-88c2-4a2524a9c042)

Thoughts @erikalogie ?

erikalogie commented:
> Thoughts @erikalogie ?

I think the way you've done it looks good

r41ph commented on web-marketplace/src/components/organisms/Order/Order.test.tsx:
```diff
@@ -1,39 +1,54 @@
 import { USD_DENOM, USDC_DENOM } from 'config/allowedBaseDenoms';
-import { describe, expect, it } from 'vitest';
+import { describe, expect, it, vi } from 'vitest';
```
We shouldn't need to import the `vitest` methods anymore, they should be globally accessible now.
